software/base/src/main/java/brooklyn/entity/basic/AbstractSoftwareProcessWinRmDriver.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    @Override
+    public String getInstallDir() {
+        // TODO: This needs to be tidied, and read from the appropriate flags (if set)
+        return "$HOME\\brooklyn-managed-processes\\installs\\" + entity.getApplicationId() + "\\" + getEntityVersionLabel()+"_"+entity.getId();
+    }
+
+    @Override
+    public int copyResource(Map<Object, Object> sshFlags, String source, String target, boolean createParentDir) {
+        if (createParentDir) {
+            createDirectory(getDirectory(target), "Creating resource directory");
+        }
+        return copyTo(new File(source), new File(target));
+    }
+
+    @Override
+    public int copyResource(Map<Object, Object> sshFlags, InputStream source, String target, boolean createParentDir) {
+        if (createParentDir) {
+            createDirectory(getDirectory(target), "Creating resource directory");
+        }
+        return copyTo(source, new File(target));
+    }
+
+    @Override
+    protected void createDirectory(String directoryName, String summaryForLogging) {
+        getLocation().executePsScript("New-Item -path \"" + directoryName + "\" -type directory -ErrorAction SilentlyContinue");
+    }
+
+    protected WinRmToolResponse executeCommand(ConfigKey<String> regularCommandKey, ConfigKey<String> powershellCommandKey, boolean allowNoOp) {
+        String regularCommand = getEntity().getConfig(regularCommandKey);
+        String powershellCommand = getEntity().getConfig(powershellCommandKey);
+        if (Strings.isNullOrEmpty(regularCommand) && Strings.isNullOrEmpty(powershellCommand)) {
+            if (allowNoOp) {
+                return new WinRmToolResponse("", "", 0);
+            } else {
+                throw new IllegalStateException(String.format("Exactly one of %s or %s must be set", regularCommandKey.getName(), powershellCommandKey.getName()));
+            }
+        } else if (!Strings.isNullOrEmpty(regularCommand) && !Strings.isNullOrEmpty(powershellCommand)) {
+            throw new IllegalStateException(String.format("%s and %s cannot both be set", regularCommandKey.getName(), powershellCommandKey.getName()));
+        }
+
+        if (Strings.isNullOrEmpty(regularCommand)) {
+            return getLocation().executePsScript(ImmutableList.of(powershellCommand));
+        } else {
+            return getLocation().executeScript(ImmutableList.of(regularCommand));
+        }
+    }
+
+    public int execute(List<String> script) {
+        return getLocation().executeScript(script).getStatusCode();
+    }
+
+    public int executePowerShellNoRetry(List<String> psScript) {
+        return getLocation().executePsScriptNoRetry(psScript).getStatusCode();
+    }
+
+    public int executePowerShell(List<String> psScript) {
+        return getLocation().executePsScript(psScript).getStatusCode();
+    }
+
+    public int copyTo(File source, File destination) {
```

Comment:
As commented in WinRmMachineLocation.java, not sure should use `File destination`, rather than `String destination`, because the OS of the brooklyn server and target VM may well be different. Thoughts?